src/ln/channelmanager.rs

```diff
@@ -785,11 +785,10 @@ impl ChannelManager {
 						},
 						Err(e) => {
 							mem::drop(channel_state);
-							add_pending_event!(events::Event::DisconnectPeer {
+							add_pending_event!(events::Event::HandleError {
 								node_id: chan.get_their_node_id(),
-								msg: if let Some(msgs::ErrorAction::DisconnectPeer { msg } ) = e.action { msg } else { None },
+								action: e.action.unwrap(),
```

Comment:
Can we make Event::HandleError::action an Option<> for now so we don't have to sprinkle all these unwrap()s everywhere? When HandleError::action switches to being non-Optional we can swap them both at the same time.

Reply:
Done on ba211a8